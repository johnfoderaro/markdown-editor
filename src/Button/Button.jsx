import React from 'react';
import { hot } from 'react-hot-loader';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Block = styled.button`
  background-color: transparent;
  border: none;
  cursor: pointer;
  font: inherit;
  text-decoration: underline;
  display: inline;
  margin: 0;
  padding: 0;
  &:hover,
  &:focus {
    text-decoration: none;
  }
`;

const Button = ({ children, ...props }) => (
  <Block {...props}>
    {children}
  </Block>
);

Button.propTypes = { children: PropTypes.string.isRequired };

export default hot(module)(Button);
