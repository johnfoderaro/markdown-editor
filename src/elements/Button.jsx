import React from 'react';
import { hot } from 'react-hot-loader';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledButton = styled.button`
  margin-top: 1rem;
  margin-bottom: 1rem;
  height: 3rem;
  width: 6rem;
  background: none;
  border: none;
  border-radius: .5rem;
  font-size: 1.25em;
  color: rgb(0, 0, 0);
  cursor: pointer;
`;

const Button = ({ children, type, ...props }) => (
  <StyledButton {...props} type={type}>
    {children}
  </StyledButton>
);

Button.propTypes = {
  children: PropTypes.node.isRequired,
  type: PropTypes.string.isRequired,
};

export default hot(module)(Button);
