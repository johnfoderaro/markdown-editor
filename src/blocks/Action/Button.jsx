import styled from 'styled-components';

import Button from '../../elements/Button';

const StyledButton = styled(Button)`
  color: ${({ type }) => (type !== 'inverse' ? 'inherit' : 'rgb(255, 255, 255)')};
  margin-right: ${({ type }) => {
    switch (type) {
      case 'new':
        return '.5rem';
      case 'save':
        return '.5rem';
      default:
        return '0';
    }
  }};
  margin-left: ${({ type }) => {
    switch (type) {
      case 'delete':
        return '.5rem';
      default:
        return '0';
    }
  }};
  border: 2px solid ${({ type }) => {
    switch (type) {
      case 'new':
        return 'rgb(25, 105, 179)';
      case 'save':
        return 'rgb(25, 179, 63)';
      case 'delete':
        return 'rgb(179, 56, 25)';
      case 'inverse':
        return 'rgb(255, 255, 255)';
      default:
        return '#eee';
    }
  }};
    &:hover {
      border: none;
      background: ${({ type }) => {
    switch (type) {
      case 'new':
        return 'rgb(25, 105, 179)';
      case 'save':
        return 'rgb(25, 179, 63)';
      case 'delete':
        return 'rgb(179, 56, 25)';
      default:
        return 'rgb(255, 255, 255)';
    }
  }};
    color: ${({ type }) => (type === 'inverse' ? 'rgb(0, 0, 0)' : 'rgb(255, 255, 255)')};
  }
`;

export default StyledButton;
