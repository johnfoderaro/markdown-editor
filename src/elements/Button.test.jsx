import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

let tree;

beforeAll(() => {
  tree = shallow(
    <Button type="new">
      Test
    </Button>,
  );
});

describe('<Button />', () => {
  it('should be defined', () => {
    expect(tree).toBeDefined();
  });
  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});
