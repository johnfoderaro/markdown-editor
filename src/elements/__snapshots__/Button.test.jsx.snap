// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    type="new"
  >
    Test
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Test",
      "type": "new",
    },
    "ref": null,
    "rendered": "Test",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  margin-top: 1rem;
  margin-bottom: 1rem;
  height: 3rem;
  width: 6rem;
  background: none;
  border: none;
  border-radius: .5rem;
  font-size: 1.25em;
  color: rgb(0, 0, 0);
  cursor: pointer;
",
        ],
      },
      "displayName": "styled.button",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "button",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Test",
        "type": "new",
      },
      "ref": null,
      "rendered": "Test",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  margin-top: 1rem;
  margin-bottom: 1rem;
  height: 3rem;
  width: 6rem;
  background: none;
  border: none;
  border-radius: .5rem;
  font-size: 1.25em;
  color: rgb(0, 0, 0);
  cursor: pointer;
",
          ],
        },
        "displayName": "styled.button",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "button",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
